functions/src/triggers/onUserCreation.ts

```diff
@@ -3,11 +3,49 @@ import * as jsonWebToken from 'jsonwebtoken'
 import { addTokenToUser } from '../db/addTokenToUser'
 import { getJWTEnv } from '../firebase/env'
 import { updateUserDb } from '../db/userDb'
-import { SUBSCRIPTIONS } from '../db/constants'
+import { SUBSCRIPTIONS, TEST_ACCOUNTS } from '../db/constants'
+import * as admin from 'firebase-admin'
+
+const makeUserData = (
+    isSubscriptionActive: boolean,
+    isQuotaReached: boolean
+) => ({
+    subscription: {
+        isActive: isSubscriptionActive,
+        isQuotaReached,
+        type: SUBSCRIPTIONS.manual,
+    },
+    usage: {
+        sentSMSs: 0,
+        sentEmails: 0,
+    },
+})
+
+const getUserData = (user: admin.auth.UserRecord) => {
+    let userData
+    if (process.env.FIREBASE_AUTH_EMULATOR_HOST) {
+        switch (user.email) {
+            case TEST_ACCOUNTS.paidUser.email:
+                userData = makeUserData(true, false)
```

Comment:
return directly rather than mutating

Reply:
clean code: a function should rather have a single success exit point

Reply:
yes but what about immutability?

Reply:
I guess it is a compromise to accept in order to increase readability in this case ;)

Reply:
what about reducer? they all return directly within a switch

Reply:
good point. We may consider [refactoring that](https://medium.com/better-programming/write-cleaner-reducers-1c433a94a166)